src/node/rpc/member_frontend.h

```diff
@@ -387,6 +394,68 @@ namespace ccf
         caller_id, {cose_sign1.begin(), cose_sign1.end()});
     }
 
+    ProposalSubmissionStatus is_proposal_submission_acceptable(
+      kv::Tx& tx,
+      const std::string& created_at,
+      const std::vector<uint8_t>& request_digest,
+      const ccf::ProposalId& proposal_id,
+      ccf::ProposalId& colliding_proposal_id,
+      std::string& min_created_at)
+    {
+      auto cose_recent_proposals = tx.rw(network.cose_recent_proposals);
+      auto key = fmt::format("{}:{}", created_at, ds::to_hex(request_digest));
+
+      std::vector<std::string> replay_keys;
+      cose_recent_proposals->foreach_key(
+        [&replay_keys](const std::string& replay_key) {
+          replay_keys.push_back(replay_key);
+          return true;
+        });
+
+      std::sort(replay_keys.begin(), replay_keys.end());
+
+      // New proposal must be more recent than median proposal kept
+      if (!replay_keys.empty())
+      {
+        min_created_at = std::get<0>(
+          nonstd::split_1(replay_keys[replay_keys.size() / 2], ":"));
+        auto [key_ts, __] = nonstd::split_1(key, ":");
+        if (key_ts < min_created_at)
+        {
+          return ProposalSubmissionStatus::TooOld;
+        }
+      }
+
+      if (cose_recent_proposals->has(key))
+      {
+        colliding_proposal_id = cose_recent_proposals->get(key).value();
+        return ProposalSubmissionStatus::DuplicateInWindow;
+      }
+      else
+      {
+        size_t window_size = 100;
```

Comment:
```suggestion
        size_t default_window_size = 100;
```

This value should probably live in `governance_history.h` too?

Reply:
It could do too, it's not going to much reuse, so I don't think it will help anything but it also won't hurt.

Reply:
@jumaffre moved to src/service/tables/config.h, which I think is a more suitable central location considering this is effectively a default on the table.